lightning/src/onion_message/router.rs

```diff
@@ -0,0 +1,248 @@
+use bitcoin::secp256k1::PublicKey;
+
+use ln::msgs::{ErrorAction, LightningError};
+use routing::gossip::{NetworkGraph, NodeId};
+use util::logger::{Level, Logger};
+
+use alloc::collections::BinaryHeap;
+use core::hash::Hash;
+use core::ops::Deref;
+use prelude::*;
+
+/// Find a path for sending an onion message.
+pub fn find_path<L: Deref, GL: Deref>(
+	our_node_pubkey: &PublicKey, receiver_pubkey: &PublicKey, network_graph: &NetworkGraph<GL>, first_hops: Option<&[&PublicKey]>, logger: L
+) -> Result<Vec<PublicKey>, LightningError> where L::Target: Logger, GL::Target: Logger
```

Comment:
So we wouldn't reuse all the information we're learning and storing in our `ProbabilisticScorer`. I can see how we're limited with the current penalty being based on the link-level and here we might be interested by node-level reliability in the path construction. That said, we can also assume that a reliable channel == a reliable onion message communication channel and go with it. I don't know if the spec says anything here or what are the thinking of other implementations ?